.github/workflows/ci.yaml

```diff
@@ -20,3 +20,22 @@ jobs:
       - run: yarn install --frozen-lockfile
       - run: yarn build
       - run: yarn test
+
+  e2e:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      # Setup gcloud application default credentials. While the credentials are not actually
+      # used because Google api services are stubbed, instantiating any of the Google node
+      # clients requires the credentials file to exist and be valid.
+      - uses: "google-github-actions/auth@v2"
+        with:
+          credentials_json: "${{ secrets.GCP_CREDENTIALS }}"
+      - uses: google-github-actions/setup-gcloud@v2
+        with:
+          version: ">= 363.0.0"
+      - run: yarn install --frozen-lockfile
+      - run: yarn e2e
```

Comment:
If you are interested, I have [a GitHub action that joins workflow jobs](https://github.com/cgrindel/gha_join_jobs) so that the branch protection rule can be configured to track a single check ([usage example](https://github.com/cgrindel/rules_swift_package_manager/blob/eba808a44392712b03b6f8145474659c7f356330/.github/workflows/ci.yml#L19-L28)). Otherwise, you need to update the branch protection rules whenever a job is added/removed from the CI workflow.

Reply:
Neat. I added it.